rtos/Thread.cpp

```diff
@@ -168,7 +168,7 @@ Thread::State Thread::get_state() {
     _mutex.lock();
 
     if (_tid != NULL) {
-        state = _obj_mem.state;
+        state = osThreadGetState(_tid);
```

Comment:
That's possibly incorrect, if we want to preserve compatibility with pre CMSIS5 code, the osThreadGetState compresses statuses to short list, RTOS C++ API have quite extensive list. I personally agree we don't need all this detailed states like ('waiting for semaphore' as opposed to 'Waiting') but that's what we used to return.

Reply:
To be backward compatible if there is no strong reason to break this. In this case it is not?